### ISSUE TITLE
Add options to create RabbitMQ lazy queue

### ISSUE THREAD

**Issue by YuhuaDeng** [dotnetcore/CAP#772]
在创建queue的时候, 如何指定 rabbitmq的 队列为lazy队列呢。 比如下面的这部分。。 查了好多文档，没有看到又lazy设置。

`
services.AddCap(options =>
 {
                    options.UseMySql(conf =>
                    {
                        conf.ConnectionString = configuration.Get<ConfigSettings>().MySQLConnectionString;
                    });
               
}
`

谢谢。

**Comment by yang-xiaodong** [comment]
Hello,

I know that Lazy mode queue by RabbitMQ is used to move messages to disk as early as practically possible,  this comes at a cost of increased disk I/O.

In this case, I have a question why don’t you choose kafka and kafka retains large amounts of data with very little overhead, it is designed for holding and distributing large volumes of messages? (this question has nothing to do with CAP)

**Comment by YuhuaDeng** [comment]
@yang-xiaodong 


it's because in our case , there are massive messages coming in a very short time. but only once per day,   I also like to replace rabbitmq with Kafka. but it's already in prod . I need to fix this asap.  

I am not familiar with CAP  but our developer mentioned me that the queue is created by CAP automatically, he can't make changes in queue. I just wish wish to add a parameter into the queue when CAP create queue.. Is it doable?

Thank you xiaodong

**Comment by yang-xiaodong** [comment]
OK, I think you can update queue model to lazy with the following command manually.

```
rabbitmqctl set_policy lazy "<replace-to-match-pattern>" '{"queue-mode":"lazy"}'
```

**Comment by YuhuaDeng** [comment]
I know above method, but thanks a lot 

So is it not possible to add parameter to code make this changes . the ideal solution might be adding parameter to it when CAP create the queue?

**Comment by yang-xiaodong** [comment]
Would like to submit a PR?

**Comment by yang-xiaodong** [comment]
Fixed in v5.0
